fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/option/GenerateOptionsBuilder.java

```diff
@@ -529,15 +523,8 @@ public GenerateOptions build() {
 			defaultNullInjectGenerator = defaultNullInjectGeneratorOperator.apply(defaultNullInjectGenerator);
 		}
 
-		int defaultArbitraryContainerMaxSize = defaultIfNull(
-			this.defaultArbitraryContainerMaxSize,
-			() -> GenerateOptions.DEFAULT_ARBITRARY_CONTAINER_MAX_SIZE
-		);
-		ArbitraryContainerInfo defaultArbitraryContainerInfo =
-			defaultIfNull(
-				this.defaultArbitraryContainerInfo,
-				() -> new ArbitraryContainerInfo(0, defaultArbitraryContainerMaxSize)
-			);
+		ArbitraryContainerInfoGenerator defaultArbitraryContainerInfoGenerator =
+			this.defaultArbitraryContainerInfoGenerator;
```

Comment:
If `defaultArbitraryContainerInfo` is not set (null), you should set `defaultArbitraryContainerInfoGenerator`.
ArbitraryContainerInfo in `ContainerPropertyGeneratorContext` means  a result of `size` manipulation.

You should set ArbitraryContainerInfoGenerator returns defaultArbitraryContainerInfo.
check out below.
```
ArbitraryContainerInfo defaultArbitraryContainerInfo =
			defaultIfNull(
				this.defaultArbitraryContainerInfo,
				() -> new ArbitraryContainerInfo(0, defaultArbitraryContainerMaxSize)
			);
```

Reply:
I don't understand `ArbitraryContainerInfoGenerator returns defaultArbitraryContainerInfo.`

As you said, I removed `defaultArbitraryContainerInfo` field in `GenerateOptions`. So, we don't have to deal with the `defaultArbitraryInfo` field in `GenerateOptionsBuilder` from my point of view 🤔 

And to get `ArbitraryInfo` returned by `ArbitraryContainerInfoGenerator.generate()`, `ContainerPropertyGeneratorContext` is required, but there seems to be no place to use it inside the `build()` method.

Reply:
And to use the code you commented, it seems that the `defaultArbitraryContainerInfo` and `defaultArbitraryContainerMaxSize` fields should be present as fields of `GenerateOptionsBuilder`, but these two fields have also been removed from `GenerateOptions`.

It's clear that I'm not understanding it properly 😭

Reply:
Yes, there is no `defaultArbitraryContainerInfo` option now.
I mean `defaultArbitraryContainerInfo` as an object of `ArbitraryContainerInfo`.

Interface `ArbitraryContainerInfoGenerator` should return `ArbitraryContainerInfo`.
```java
@FunctionalInterface
public interface ArbitraryContainerInfoGenerator {
	ArbitraryContainerInfo generate(ContainerPropertyGeneratorContext context);
}
```

Since it is functional interface, you don't need `ContainerPropertyGeneratorContext`.
You could use it by lambda expression. 
such as `(ctx) -> new ArbitraryContainerInfo(0, 3)`

`defaultArbitraryContainerInfo` option has a default `ArbitraryContainerInfo` like below.
`defaultArbitraryContainerInfoGenerator` option should have a default `defaultArbitraryContainerInfoGenerator`, same as deprecated `defaultArbitraryContainerInfo` option.

```java
new ArbitraryContainerInfo(0, defaultArbitraryContainerMaxSize)
```

So you have to deal with the situation user does not use `defaultArbitraryContainerInfoGenerator` option.

Reply:
Thanks for the clear and detailed explanation 👍 
In commit [32dd7be](https://github.com/naver/fixture-monkey/pull/581/commits/32dd7be3838cc34c62052fb10a6c8e182ecd45e4), I tried again according to your guide.